src/class.c

```diff
@@ -1634,22 +1661,27 @@ bool InterfaceDeclaration::isBaseOf(ClassDeclaration *cd, int *poffset)
     {
         BaseClass *b = cd->interfaces[j];
 
-        //printf("\tbase %s\n", b->base->toChars());
-        if (this == b->base)
+        //printf("\tX base %s\n", b->sym->toChars());
+        if (this == b->sym)
         {
             //printf("\tfound at offset %d\n", b->offset);
             if (poffset)
             {
                 *poffset = b->offset;
-                if (j && cd->isInterfaceDeclaration())
+                // The conversion is interface(cd) to interface(this).
+                if (cd->isInterfaceDeclaration())
```

Comment:
Ah, perhaps I was wrong? That is necessary to succeed the last assert of this code:

```d
interface IA      { int mA(); }
interface IB : IA { int mB(); }
interface IC : IB { }

interface ID : IC { int mD(); }
class C : IA, ID
{
    int mA() { return 1; }
    int mB() { return 2; }
    int mD() { return 3; }
}

void main()
{
    C c = new C();
    IA ia = c;
    ID id = c;
    IA ia2 = id;
    assert(ia2 is ia);
```

However the change is not necessary for both `ia.mA()` and `ia2.mA()` calls...